### ISSUE TITLE
java.lang.NoSuchMethodError: org.apache.tomcat.jni.SSLContext.setCACertificateBio(JJ)Z

### ISSUE THREAD

**Issue by zmyer** [netty/netty#6219]
### Actual behavior
java.lang.NoSuchMethodError: org.apache.tomcat.jni.SSLContext.setCACertificateBio(JJ)Z
	at io.netty.handler.ssl.ReferenceCountedOpenSslServerContext.newSessionContext(ReferenceCountedOpenSslServerContext.java:162)
	at io.netty.handler.ssl.OpenSslServerContext.<init>(OpenSslServerContext.java:349)
	at io.netty.handler.ssl.OpenSslServerContext.<init>(OpenSslServerContext.java:334)
	at io.netty.handler.ssl.SslContext.newServerContextInternal(SslContext.java:414)
	at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:402)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
### Netty version
4.1.6.Final
### JVM version (e.g. `java -version`)
jdk1.8
### OS version (e.g. `uname -a`)
Mac OSX

**Comment by slandelle** [comment]
This information you provided is incorrect.Your stacktrace says there's a call to `org.apache.tomcat.jni.SSLContext.setCACertificateBio` in `ReferenceCountedOpenSslServerContext` on line 162, but there's [no such call in Netty 4.1.6](https://github.com/netty/netty/blob/netty-4.1.6.Final/handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslServerContext.java#L162) that you claim to be using.

Actually, the `setCACertificateBio` only exists [since Netty 4.1.7](https://github.com/netty/netty/blob/netty-4.1.7.Final/handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslServerContext.java#L162) an this method was [introduced in netty-tcnative 1.1.33.Fork25](https://github.com/netty/netty-tcnative/commit/9d9a37c1f9b38104801be463244d4be587d20b7e#diff-b47b55d3907d27bd4cb6e1da17580478).

Basically, you're probably using Netty 4.1.7, and your netty-tcnative version is not up-to-date.

**Comment by normanmaurer** [comment]
Exactly like @slandelle said :)
